packages/mdc-textfield/_mixins.scss

```diff
@@ -20,35 +20,398 @@
+  // postcss-bem-linter: end
+
+  // mdc-form-field tweaks to align text field label correctly
+  // stylelint-disable selector-max-type
+  .mdc-form-field > .mdc-text-field + label {
+    @include mdc-feature-targets($feat-structure) {
+      align-self: flex-start;
+    }
+  }
+  // stylelint-enable selector-max-type
+
+  .mdc-text-field--focused {
+    @include mdc-text-field-focused_($query);
+  }
+
+  .mdc-text-field--invalid {
+    @include mdc-text-field-invalid_($query);
+  }
+
+  .mdc-text-field--textarea.mdc-text-field--invalid {
+    @include mdc-text-field-textarea-invalid_($query);
+  }
+
+  .mdc-text-field--disabled {
+    @include mdc-text-field-disabled_($query);
+  }
+
+  .mdc-text-field--textarea.mdc-text-field--disabled {
+    @include mdc-text-field-textarea-disabled_($query);
+  }
+
+  .mdc-text-field--end-aligned {
+    @include mdc-text-field-end-aligned_($query);
+  }
+
+
+  @include mdc-floating-label-shake-keyframes(
+    text-field-dense,
+    $mdc-text-field-dense-label-position-y,
+    0%,
+    $mdc-text-field-dense-label-scale,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined,
+    $mdc-text-field-outlined-label-position-y,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined-dense,
+    $mdc-text-field-outlined-dense-label-position-y,
+    0%,
+    $mdc-text-field-dense-label-scale,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined-leading-icon,
+    $mdc-text-field-outlined-label-position-y,
+    $mdc-text-field-outlined-with-leading-icon-label-position-x,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined-leading-icon-dense,
+    $mdc-text-field-outlined-dense-label-position-y,
+    $mdc-text-field-outlined-dense-with-leading-icon-label-position-x,
+    $mdc-text-field-dense-label-scale,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined-leading-icon-rtl,
+    $mdc-text-field-outlined-label-position-y,
+    -$mdc-text-field-outlined-with-leading-icon-label-position-x,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    text-field-outlined-leading-icon-dense-rtl,
+    $mdc-text-field-outlined-dense-label-position-y,
+    -$mdc-text-field-outlined-dense-with-leading-icon-label-position-x,
+    $mdc-text-field-dense-label-scale,
+    $query: $query
+  );
+  @include mdc-floating-label-shake-keyframes(
+    textarea,
+    $mdc-text-field-textarea-label-position-y,
+    0%,
+    $query: $query
+  );
+}
+
+// This API is intended for use by frameworks that may want to separate the ripple-related styles
+// from the other text field styles. It is recommended that most users use `mdc-text-field-core-styles` instead.
+@mixin mdc-text-field-ripple($query: mdc-feature-all()) {
+  @include mdc-ripple-common($query);
```

Comment:
This mixin wasn't included before. Remove this.

Reply:
Wasn't the ripple previously included via an import? I think that this is what we did for the other components where we separated the ripple styles from everything else.

Reply:
Can we `@import` ripple/common in mdc-text-field file instead? (Similar to floating label, line ripple etc.)

It is always safer to retain the CSS ordering to avoid any further breaking changes.

Reply:
It goes against the pattern we've set until now though. Here are a few examples:

https://github.com/material-components/material-components-web/blob/master/packages/mdc-button/_mixins.scss#L138
https://github.com/material-components/material-components-web/blob/master/packages/mdc-checkbox/_mixins.scss
https://github.com/material-components/material-components-web/blob/master/packages/mdc-radio/_mixins.scss

If you still think that it's okay to diverge in this case it's fine by me.

Reply:
hmm.. Can you add `// COPYBARA_COMMENT_THIS_LINE` comment next to the -common mixin then? It'll create duplicate styles internally otherwise.